### PR TITLE
Exposing dnsPolicy and dnsConfig on the Strimzi api PodTemplate to allow custom name resolution for Pod resources

### PR DESCRIPTION
**Pull request by gertschouten**
### Type of change

- Enhancement / new feature

### Description

By appending Strimzi's common PodTemplate in the Strimzi api with additional properties dnsPolicy and dnsConfig the system configuration file /etc/recolv.conf for Strimzi Pod resources can be controlled, thereby deciding how domain name resolution should be performed on a resource level. 
Strimzi's CRDs will thereby allow user input for name resolution customization.
WorkloadUtils in the Strimzi cluster operator for PodBuilder and PodTemplateSpecBuilder propagate these properties to Pods controlled by the operator.
Provisioned Pods will thereby utilize name resolution configuration based on the user input if specified and fall back to defaults, similar to the current situation, in case no dnsConfig or dnsPolicy is specified by the user. 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [X] Update documentation
- [x] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md
- [x] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by scholzj** [comment]
You should update the title and the description of the PR. You also seem to have some bad commits in there. The Pr has to be done against the `main` branch with the code from there. That might be also what broker the DCO signoff.

File: api/src/main/java/io/strimzi/api/kafka/model/common/template/PodTemplate.java
Comment (scholzj) [inline api/src/main/java/io/strimzi/api/kafka/model/common/template/PodTemplate.java:58]: I think this should have an Enum for the supported values. Check for example how https://github.com/strimzi/strimzi-kafka-operator/blob/0c334d8100a8c0273f28e4914a4f09fc0fa0fbb1/api/src/main/java/io/strimzi/api/kafka/model/common/template/ExternalTrafficPolicy.java#L12 is used.

**Review by scholzj — COMMENTED** [review]
We in general deal with the PR only once the proposal is approved. But I left one comment. No need to address it now, you can do it only once the proposal is approved. It would be also great to fix all the unnecessary whitespace changes that make the reviews harder, but will also mess up the Git history etc. (e.g. in the `pom.xml` files, the `values.yaml`, etc.)

You will also need to run `make crd_install` to get the CRD changes copied out to the Helm Chart as well. And you will need to fix the DCO signoff. The instructions should be under the _Details_ link next to the DCO signoff status in the PR.

File: api/src/main/java/io/strimzi/api/kafka/model/common/template/PodTemplate.java
Comment (gertschouten) [reply api/src/main/java/io/strimzi/api/kafka/model/common/template/PodTemplate.java]: Addressed in [9bd12a2](https://github.com/strimzi/strimzi-kafka-operator/pull/10949/commits/9bd12a2eef18ffc7a3276b76ef101f326df94045)

**Review by scholzj — COMMENTED** [review]
I left some more nits. Also, there have been some changes to the main branch in between - could oyu please rebase it to address the conflicts? Thanks.

This might be also worth a record in the `CHANGELOG.md` file?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/WorkloadUtils.java
Comment (gertschouten) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/WorkloadUtils.java]: Addressed in [fffc5db](https://github.com/strimzi/strimzi-kafka-operator/pull/10949/commits/fffc5dbff412160661da48b8dc13942b432e9524)

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/WorkloadUtils.java
Comment (gertschouten) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/WorkloadUtils.java]: Addressed in [fffc5db](https://github.com/strimzi/strimzi-kafka-operator/pull/10949/commits/fffc5dbff412160661da48b8dc13942b432e9524)

**Review by scholzj — APPROVED** [review]
LGTM. Thanks for the PR.

File: api/src/main/java/io/strimzi/api/kafka/model/common/template/DnsPolicy.java
Comment (ppatierno) [inline api/src/main/java/io/strimzi/api/kafka/model/common/template/DnsPolicy.java:12]: Can we use the "_" in the enum values to separate works, so having CLUSTER_FIRST and CLUSTER_FIRST_WITH_HOST_NET?

File: api/src/main/java/io/strimzi/api/kafka/model/common/template/PodTemplate.java
Comment (ppatierno) [inline api/src/main/java/io/strimzi/api/kafka/model/common/template/PodTemplate.java:193]: Suggestion, "If specified, it will be merged to ..."

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/WorkloadUtils.java
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/WorkloadUtils.java:467]: What's this change?

**Review by ppatierno — APPROVED** [review]
LGTM, thanks for the PR! I left a couple of nits so when they are fixed I guess we can merge this.

**Pull request merged**
